hacspec/src/machine_integers.rs

```diff
@@ -138,157 +223,258 @@ macro_rules! implement_public_mi {
     };
 }
 
-implement_public_mi!(u8);
-implement_public_mi!(u16);
-implement_public_mi!(u32);
-implement_public_mi!(u64);
-implement_public_mi!(u128);
+implement_public_unsigned_mi!(u8, 8);
+implement_public_unsigned_mi!(u16, 16);
+implement_public_unsigned_mi!(u32, 32);
+implement_public_unsigned_mi!(u64, 64);
+implement_public_unsigned_mi!(u128, 128);
 
-implement_public_mi!(i8);
-implement_public_mi!(i16);
-implement_public_mi!(i32);
-implement_public_mi!(i64);
-implement_public_mi!(i128);
+implement_public_signed_mi!(i8, 8);
+implement_public_signed_mi!(i16, 16);
+implement_public_signed_mi!(i32, 32);
+implement_public_signed_mi!(i64, 64);
+implement_public_signed_mi!(i128, 128);
 
+// ========== Secret Machine Integers ========== //
+
+macro_rules! implement_secret_unsigned_mi {
+    ($t:ident,$base:ty,$bits:literal) => {
+        implement_secret_mi!($t, $base, $bits);
+        impl ModNumeric for $t {
+            /// (self - rhs) % n.
+            #[cfg_attr(feature="use_attributes", library(hacspec))]
```

Comment:
Is modulo arithmetic constant-time for secret integers ? Taking the modulo amounts to a division isn't it ?

Reply:
We can implement it in ct. See `ct_div` in `math_util.rs`. I didn't use it here because I wanted to do the re-org first. In a follow up all this in here should be replaced with ct code.

Reply:
Ok, perfect!